src/sklearn_evaluation/plot/clustering.py

```diff
@@ -262,21 +263,26 @@ def silhouette_analysis(
             "Cannot plot silhouette analysis ."
         )
 
+    # if no ax is passed by user generate new plot
+    # for each model
+    param_ax = ax
```

Comment:
do we need this? I think we can just use `ax`

Reply:
In the case where user doesn't pass the ax value, the loop should create a fresh `ax` everytime. However if this is passed we should use the existing one.
If I remove this check all the plots are generated one on top of the other. Noticed this while adding the `test_ax_silhouette` testcase

Reply:
On second thoughts, does user need to pass a list of `Axes` objects as the `ax` parameter? Since there are separate plots for each model 
@edublancas

Reply:
ah. good point! this is the first plot that creates multiple "plots". I think the behavior should be this:

no ax argument (i.e., ax=None): create new subplots as necessary
ax passed: verify it's a list with a length equal to the number of plots to generate. it so, then use it for the plots, otherwise, raise an error